src/Model/Gallery.php

```diff
@@ -144,8 +144,6 @@ public function getDefaultFormat()
      */
     public function setGalleryHasMedias($galleryHasMedias)
     {
-        $this->galleryHasMedias = new ArrayCollection();
-
```

Comment:
Shouldn't the collection be cleared first? There might be things to remove in there.

Reply:
I don't know why we should clear the collection first.

Deleting an image from the Gallery works just fine.

Reply:
Because anyone might call `setGalleryHasMedias` with any custom code expect a replacement to happen, not a merge.

Reply:
Why would anyone want to replace all images of an existing Gallery with a complete set of new images? In that case, he should better add a custom method like "replaceGalleryHasMedia", since he most probably will have the Gallery model extended anyways.

This issue just removed all images from my local development instance of a website. I think you should focus on fixing this before someone experiences some real trouble.

Reply:
> Why would anyone want to replace all images of an existing Gallery with a complete set of new images? In that case, he should better add a custom method like "replaceGalleryHasMedia", since he most probably will have the Gallery model extended anyways.

I don't know, you don't know, let's not take any chances, ok? I don't know why you insist on removing this, you or @gremo? The `Collection` interface has a `clear` method: https://github.com/doctrine/collections/blob/master/lib/Doctrine/Common/Collections/Collection.php#L43

Just call, let me approve this PR, and get that bug fixed.

Reply:
Oh sorry I get it now… not sure how to solve this.